lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/MavenProjectCache.java

```diff
@@ -113,6 +113,25 @@ private void check(DOMDocument document) {
 		}
 	}
 
+	/**
+	 * Parses and adds a document for a given URI into the projects cache
+	 * 
+	 * @param uri An URI of a document to add
+	 * @param document A document to add
+	 */
+	public void addDocument(URI uri, DOMDocument document ) {
```

Comment:
As mentioned above, I don't think the document should be an argument of this method as we can fully deduce it from the uri.

Reply:
As I wrote above - once the document is already created in `visitFile()` having this parameter here allows to reuse it instead of creating it once again by URI inside `addDocument()` method.

Reply:
What makes it necessary to create the document in visitFile() ?

Reply:
The purpose of this was detection of pom xml documents. 
I have moved this detection to `MavenProjectCache.addDocument(URI)`